madprops/serializers.py

```diff
@@ -61,95 +88,123 @@ class Property(model):
+            prop.save()
+        else:
+            prop = self.Meta.model.objects.create(**property_data)
+
+        return prop
+
+    def to_internal_value(self, data):
+        data = self._to_extended_dict(data)
 
-    _options_class = NestedPropertySerializerOptions
+        # Handle JSON fields (value encoded as JSON).
+        if data['name'] in self.opts.json_props:
+            data['value'] = json.dumps(data['value'])
 
-    def __init__(self, **kwargs):
-        super(NestedPropertySerializer, self).__init__(**kwargs)
-        self.many = True
+        data = self._add_parent_obj_field(data)
+        return super(PropertySerializer, self).to_internal_value(data)
 
-    def _from_native_hook(self, data):
+    def _to_extended_dict(self, data):
+        """ Convert dictionary of properties:
+        {<prop_name>: <prop_value>} ->
+            {'name': <prop_name>, 'value': <prop_value>}
+        """
+        prop_name, prop_value = data.items()[0]
+        return {'name': prop_name, 'value': prop_value}
+
+    def _add_parent_obj_field(self, data):
+        # Property requires ID of parent object (properties owner, e.g. User
+        # for user's properties case). Take it from context, which is passed
+        # from view.
+        parent_obj_field = self.opts.parent_obj_field
+        parent_id_field = parent_obj_field + '_id'
+        data[parent_obj_field] = self.context.get(
+            'parent_id') or self.context['view'].kwargs.get(
+                parent_id_field)
         return data
+
+
+class PropertiesOwnerSerializer(ModelSerializer):
+    """Base class for "parent" serializers.
+
+    If you want to use PropertySerializer with a parent model (e.g. as a
+    nested serializer for User serializer), and have this field writable  -
+    you have to inherit you parent serializer from this class, not
+    ModelSerializer.
+    """
+    def create(self, validated_data):
+        # Standard DRF .create() prohibits nested writable serializers, so
+        # we do a trick - first remove properties from validated data, save
+        # parent object, and then save properties separately.
+        validated_data_minus_properties = dict(validated_data)
+
+        # We can theoretically have more than one properties field.
+        properties_fields = []
+        for field in validated_data:
+            if self._is_properties_field(field):
+                del validated_data_minus_properties[field]
+                properties_fields.append(field)
+
+        instance = super(PropertiesOwnerSerializer, self).create(
+            validated_data_minus_properties)
+
+        # Now save properties separately.
+        [self._save_properties(instance, field)
+            for field in properties_fields]
+
+        return instance
+
+    def _is_properties_field(self, field_name):
+        serializer = self.fields[field_name]
+        return isinstance(serializer, ListSerializer) and isinstance(
+            serializer.child, PropertySerializer)
```

Comment:
Right, but we know for sure, that if we have field with `ListToDictSerializer` type, it's something that have `PropertySerializer` as child.  Or we plan to use `ListToDictSerializer` for something other than `PropertySerializer`?